src/Forms/GridField/GridFieldPrintButton.php

```diff
@@ -126,21 +126,26 @@ public function getURLHandlers($gridField)
      */
     public function handlePrint($gridField, $request = null)
     {
+        $orig = ini_get('max_execution_time');
         set_time_limit(60);
         Requirements::clear();
 
         $data = $this->generatePrintData($gridField);
 
         $this->extend('updatePrintData', $data);
 
+        $ret = null;
         if ($data) {
-            return $data->renderWith([
+            $ret = $data->renderWith([
                 get_class($gridField) . '_print',
                 GridField::class . '_print',
             ]);
         }
+        // Reset the time limit, this is done mainly to prevent the 60 second time limit from
+        // making unrelated unit-tests run out of execution time when running a testsuite.
+        set_time_limit($orig);
```

Comment:
Use a `try/finally` - if there's exceptions before resetting the limit we still want to set it back to normal.

Reply:
I will note that calling this _resets_ the time limit. According to [the PHP docs](https://www.php.net/manual/en/function.set-time-limit.php) "The default limit is 30 seconds or, if it exists, the max_execution_time value defined" and "When called, set_time_limit() restarts the timeout counter from zero".

So calling this here could result in giving _more time_ than was intended.

If this is _only_ being done to resolve something in tests, then something should be done in `SapphireTest`, not here.